### PR TITLE
Submitting before work.

### PR DESCRIPTION
**Pull request by jacob-berger**
Will add more tests shortly.

### CODE REVIEW

File: Assignment6/Array.cs [hunk 1]
```diff
@@ -1,10 +1,121 @@
...
+            if (array is null)
+            {
+                throw new ArgumentNullException(nameof(array), "Array is null");
+            }
+
+            if (arrayIndex < 0 || arrayIndex > Capacity)
+            {
+                throw new IndexOutOfRangeException("Index out of bounds");
+            }
+
```
Comment (Cybotive) [inline Assignment6/Array.cs]: Maybe add this from the ICollection\<T\> docs:
"
ArgumentException

The number of elements in the source ICollection\<T\> is greater than the available space from arrayIndex to the end of the destination array.
" -https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.icollection-1.copyto?view=netframework-4.8

File: Assignment6Tests/ArrayTests.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        public void AddTest()
+        {
+            //Arrange
+            Array<String> array = new Array<String>(10);
+
+            //Act
+            array.Add("Item");
+
+            //Assert
+            Assert.AreEqual("Item", array.ToString());
```
Comment (Cybotive) [inline Assignment6Tests/ArrayTests.cs]: .ToString() doesnt seem like it'd return "Item" in the current state of Array\<T\>. Maybe use .Contains() or array[0] when implemented.

File: Assignment6/ArrayCollection.cs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            {
+                return false;
+            }
+
+            Collection.Remove(item);
+
+            return false;
+        }
+
+        public IEnumerator<T> GetEnumerator()
```
Comment (MarkMichaelis) [inline Assignment6/ArrayCollection.cs]: Wait.. can't you just use:
```
public IEnumerator<T> GetEnumerator() => Collection.GetEnumerator();
```

File: Assignment6/ArrayCollection.cs [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        {
+            foreach(T item in Collection)
+            {
+                yield return item;
+            }
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
```
Comment (MarkMichaelis) [inline Assignment6/ArrayCollection.cs]: Consider using an expression bodied member.

File: Assignment6Tests/ArrayTests.cs [hunk 5]
```diff
@@ -0,0 +1,245 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Assignment6;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Assignment6.Tests
+{
+    [TestClass()]
```
Comment (MarkMichaelis) [inline Assignment6Tests/ArrayTests.cs]: Why is there not test of iterating over the class using foreach?

**Comment by MarkMichaelis** [comment]
##  (jacob-berger)

## Assignment6

| Build/Compile (Build Succeeded: 0 Error(s), **12 Warning(s)**)| **<10>** |
| --------------------- | ------ |
| Assignment compiles even though major pieces might be missing, or there are significant problems: **✔** |  **5** |
| The assignment contains only a few required elements or has significant problems. | **5** |


| Assignment  | **15** |
| --------------------- | ------ |
| Supports collecting a fixed number of items once instantiated. | ✔ |
| Has a constructor via which you can specify fixed width.  (Other constructors are welcome.) | ✔ |
| Handle null appropriately when items are missing (this is likely a judgement call so justify your answer in the comments.) | ✔ |
| Throw an exception if an item doesn't exist. | ✔ |
| Include an index operator. | ❌ |
| Support using foreach over the items in the `Array<T>` class. | ✔ |
| Support a collection initializer. | ✔ |
| Implement `System.Collections.Generic.ICollection<T>`. | ✔ |
| Ignore support for multi-dimensional arrays. | ✔ |
| Add a `Capacity` property to return the capacity (the value specified in the constructor) of the collection. | ✔ |


| Project Analysis |   |
| -----------------| - |
| Nullable Enabled: | **❌** |
| CodeReview Performed | **✔** |
| Add the code analysis packages and appropriately handle all warnings* |  **✔** |
*<sub><sup>`IntelliTectAnalyzer.dll`,`Microsoft.NetCore.Analyzers.dll`,`Microsoft.CodeQuality.Analyzers.dll`,`Microsoft.NetCore.Analyzers.dll`,`Microsoft.NetFramework.Analyzers.dll`. (It is okay to appropriately disable rules such as  localization/culture rules.)</sub></sup>


| ProjectName | Nullable | Solutions | Analyzers | Disabled Warnings | Waning Level |
| -- | -- | -- | -- | -- | -- |
| Assignment6 | Enable | Assignment6.sln |  | CS1701, CS1702, CS2008 | 4 |
| Assignment6Tests | Disable | Assignment6.sln |  | CS1701, CS1702 | 4 |


| Testing | **15** |
| --------------------- | ------ |
| Test Run Succeeded: Total tests=19, Passed=19, Failed=0, Skipped=0 | **82%** |
| Unit test project fails to compile, most test fails, or contains significant problems. | **10** |
| Most assigned work is tested or contain some errors. | **5** |
| All assigned work is tested | **5** |


| Extra Credit | **0** |
| --------------------- | ------ |
| Write a cast operator to a C# array or a `System.Collections.Generic.List<T>`. | ❌ |
| Add support for an `IReadOnlyArray` interface (for you to define) that provides readonly access to the count and each element in the array.  Be sure to appropriately decorate it for covariance and contravariance | ❌ |

**Pull request closed without merging**